src/context-parser.js

```diff
@@ -611,55 +616,52 @@ Parser.prototype.getCurrentState = function() {
 /**
  * @function Parser#getStates
  *
- * @returns {Array} An array of states.
+ * @returns {Array} An array of states, or undefined if enableStateTracking is off
  *
  * @description
  * Get the states of the HTML5 page
  *
  */
 Parser.prototype.getStates = function() {
-    return this.states.slice();
+    return this.states && this.states.slice();
 };
 
 /**
  * @function Parser#setInitState
+ * @alias Parser#reset
  *
  * @param {integer} state - The initial state of the HTML5 Context Parser.
  *
  * @description
- * Set the init state of HTML5 Context Parser.
+ * Set the init state of HTML5 Context Parser. Once this is called, any existing states will be reset, in order to make use of the init state for processing.
  *
  */
-Parser.prototype.setInitState = function(state) {
-    this.states = [state];
-    return this;
-};
+Parser.prototype.setInitState = Parser.prototype.reset;
```

Comment:
according to this proposal, the reset() function need to clear the variable states to a new array, otherwise, if the second contextualize() is processing the shorter string than the first contextualize(), the return states will be wrong, as it contains both new/old states in an array.